src/types/index.ts

```diff
@@ -0,0 +1,60 @@
+import type * as views from '@/views'
+
+export enum NavigationActionType {
+  ADD = 'add',
+  REMOVE = 'remove',
+  SET = 'setContent'
+}
+
+export type View = keyof typeof views
+
+export interface NavigationAction {
+  type: NavigationActionType
+  props?: ViewProps
+  name?: string
+  component?: React.FC<ViewProps>
+  content?: ContentState[]
+  id?: string
+}
+
+export interface RegistryItem {
+  metadata: {
+    name: string
+    path: string
+  }
+  component: React.FC<ViewProps>
+}
+
+export interface Registry {
+  get: (key: View) => RegistryItem
+  set: () => void
+}
+
+export interface NavigationState {
```

Comment:
This is sometimes used without the properties as empty object. If that is correct it should be
```
export interface NavigationState {
  registry?: Registry
  content?: JSX.Element[]
}
```